experiments/ClimaEarth/components/land/climaland_integrated.jl

```diff
@@ -0,0 +1,511 @@
+    soilco2_sources = (CL.Soil.Biogeochemistry.MicrobeProduction{FT}(),)
+    soilco2_boundary_conditions = (; top = soilco2_top_bc, bottom = soilco2_bot_bc)
+    return (;
+        boundary_conditions = soilco2_boundary_conditions,
+        sources = soilco2_sources,
+        domain = domain,
+        parameters = soilco2_ps,
+    )
+end
+
+"""
+    create_soil_args(::Type{FT}, domain, spatially_varying_soil_params)
+
+Creates the arguments for the soil model.
+"""
+function create_soil_args(::Type{FT}, domain, spatially_varying_soil_params) where {FT}
+    (;
+        ν,
+        ν_ss_om,
+        ν_ss_quartz,
+        ν_ss_gravel,
+        hydrology_cm,
+        K_sat,
+        S_s,
+        θ_r,
+        PAR_albedo_dry,
+        NIR_albedo_dry,
+        PAR_albedo_wet,
+        NIR_albedo_wet,
+        f_max,
+    ) = spatially_varying_soil_params
+
+    soil_params = CL.Soil.EnergyHydrologyParameters(
+        FT;
+        ν,
+        ν_ss_om,
+        ν_ss_quartz,
+        ν_ss_gravel,
+        hydrology_cm,
+        K_sat,
+        S_s,
+        θ_r,
+        PAR_albedo_dry = PAR_albedo_dry,
+        NIR_albedo_dry = NIR_albedo_dry,
+        PAR_albedo_wet = PAR_albedo_wet,
+        NIR_albedo_wet = NIR_albedo_wet,
+    )
+    return (domain = domain, parameters = soil_params)
+end
+
+"""
+    create_snow_args(::Type{FT}, domain, earth_param_set, dt)
+
+Creates the arguments for the snow model.
+"""
+function create_snow_args(::Type{FT}, domain, earth_param_set, dt) where {FT}
+    snow_parameters = CL.Snow.SnowParameters{FT}(dt; earth_param_set = earth_param_set)
+    return (; parameters = snow_parameters, domain = CL.obtain_surface_domain(domain))
+end
+
+###############################################################################
+### Functions required by ClimaCoupler.jl for a SurfaceModelSimulation
+###############################################################################
+
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:area_fraction}) = sim.area_fraction
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:beta}) =
+    CL.surface_evaporative_scaling(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:emissivity}) = sim.integrator.p.ϵ_sfc
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:energy}) = CL.total_energy(sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_direct_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:surface_diffuse_albedo}) =
+    CL.surface_albedo(sim.model, sim.integrator.u, sim.integrator.p)
+Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:water}) = CL.total_water(sim.integrator.u, sim.integrator.p)
+function Interfacer.get_field(sim::ClimaLandSimulation, ::Val{:height_disp})
+    d_canopy = CL.displacement_height(sim.model.canopy, sim.integrator.u, sim.integrator.p)
```

Comment:
I wonder if we could leave these in another section marked off with ###### to note that that in the current implementation, these are not used?

Reply:
seeing them may make the coupling harder for someone to understand, just a thought

Reply:
sorry I forgot to remove these but just did it! let me push now